src/js/components/DateInput/DateInput.js

```diff
@@ -244,7 +264,29 @@ Use the icon prop instead.`,
       );
     }
 
-    const calendarButton = (
+    const onClickCopy = () => {
+      global.navigator.clipboard.writeText(textValue);
+      announce(readOnlyCopyValidation, 'assertive');
+      setTip(readOnlyCopyValidation);
+    };
+
+    const onBlurCopy = () => {
+      if (tip === readOnlyCopyValidation) setTip(readOnlyCopyPrompt);
+    };
+
+    const calendarButton = readOnlyCopy ? (
```

Comment:
For clarity, should we rename this from "calendarButton" to something else? From a readability standpoint, it was a little strange in the logic below to see `readOnlyCopy && calendarButton`

Reply:
done, renamed to `DateInputButton`. I feel like the naming isn't perfect but hopefully this makes it more clear that this is just the button within the DateInput rather than the button to open the calendar.